tasks.rb

```diff
@@ -2084,6 +2085,12 @@ def apply_hpxml_modification_sample_files(hpxml_path, hpxml)
                                            water_heating_system_idref: nil, # Apply to all water heaters
                                            solar_fraction: 0.65)
     end
+    if ['base-dhw-multiple-undersized.xml'].include? hpxml_file
+      hpxml_bldg.water_heating_systems.each do |water_heating_system|
+        water_heating_system.heating_capacity = 5000 if water_heating_system.water_heater_type == HPXML::WaterHeaterTypeStorage
+        water_heating_system.temperature = 100.0 if [HPXML::WaterHeaterTypeHeatPump, HPXML::WaterHeaterTypeTankless].include?(water_heating_system.water_heater_type)
```

Comment:
This is kind of weird, the HPWH/tankless systems are not actually undersized, they just have their setpoint artificially lowered to result in unmet showers. At least for storage/HPWH, can't we just set a small tank volume instead? Maybe we don't need all the sample files. We can probably remove the tankless one, for example.

(Also, hopefully at some point in the future, we'll allow optional capacity inputs for HPWHs and tankless systems.)

Reply:
I tried instead reducing tank volume (e.g., to 1 gal). Water Heater Use Side Outlet Temperature doesn't appear to drop (even when using stochastic schedules).

Reply:
Any idea, @jmaguire1?

Reply:
Definitely a little surprised, but I know we do have some code that scale the max flow rate to ensure that you never draw out more than the whole tank volume in an hour https://github.com/NREL/OpenStudio-HPXML/blob/e770f42307069ba1ed46467ddaee3866d2f70551/HPXMLtoOpenStudio/resources/waterheater.rb#L837

I think this means the water heater might be reducing the flow rate through it with stochastic schedules if you try to specify like 2 gal/min, the WH might only see lower flow rate. E+ will throw an error if you have a flow rate on the use side larger than the tank volume/hour, so we do need this code. If reducing the tank volume and heater capacity doesn't give you unmet loads, I'm not sure that there's a better way to do this without reducing the setpoint.

Reply:
Hmm, I'm definitely confused here. The code you are referencing above is in the `apply_solar_thermal` method. Looks like `storage_tank.setUseSideDesignFlowRate(UnitConversions.convert(storage_volume, 'gal', 'm^3') / 60.1)` is used only for stratified tanks, not mixed. I'm seeing that use side outlet temperature doesn't drop for an HPXML file with storage water heater (i.e., mixed).

Reply:
Here is 40 gal vs 1 gal:
![image](https://github.com/user-attachments/assets/84fae02b-6944-4ed2-8f64-0773398730e9)

Reply:
> If reducing the tank volume and heater capacity doesn't give you unmet loads, I'm not sure that there's a better way to do this without reducing the setpoint.

I am confused as well. This new output seems meaningless if it doesn't capture unmet hours associated with not having a large enough storage tank. Isn't that exactly the kind of situation we're trying to evaluate in the context of a (lower heating capacity, potentially without backup electric resistance) HPWH?

Reply:
@jmaguire1 Any ideas here?

Reply:
So I did do some runs here looking at super small tank volumes. Here's some of the relevant results for an undersized tank (https://github.com/NREL/OpenStudio-HPXML/blob/unmet_wh_loads/workflow/sample_files/base-dhw-undersized.xml). Relevant WH info:

- 40 gal nominal volume, electric storage water heater
- 1100 Btu/hr burners
- Mixed tank model
- Stochastic draw profile

![image](https://github.com/user-attachments/assets/2a8d6911-07bd-469d-999b-da6950f35b16)

I think this is pretty much what we'd expect to see, that with the super small burner capacity, the tank temperature DOES drop low enough that you get unmet loads, and we see them being tracked when there's a shower draw and the outlet temp is < 105 F (~40.56 C)?

With just dropping the tank volume down, you can still have the WH meet the load for smaller draws. It basically acts like a tiny tankless WH (I think we actually do assume ~1 gal volume for tankless water heaters right now, but a huge burner capacity. See https://github.com/NREL/OpenStudio-HPXML/blob/32d0767c8ad44c548b3cfe6f5e5fc9553a420dea/HPXMLtoOpenStudio/resources/waterheater.rb#L106)